Bug(docker-compose): Connectors unable to fetch process definitions

The connectors are unable to fetch the process definitions from Operate. This error message appears:

```
2024-04-03 10:32:18 2024-04-03T08:32:18.741Z ERROR 1 --- [   scheduling-1] i.camunda.common.http.DefaultHttpClient  : Failed POST with responseType class io.camunda.operate.model.SearchResult, parameterType class io.camunda.operate.model.ProcessDefinition, selector io.camunda.operate.model.SearchResult<io.camunda.operate.model.ProcessDefinition>, body io.camunda.operate.search.SearchQuery@624e5206 due to Response not successful: 500
2024-04-03 10:32:18 2024-04-03T08:32:18.741Z ERROR 1 --- [   scheduling-1] o.s.s.s.TaskUtils$LoggingErrorHandler    : Unexpected error occurred in scheduled task
2024-04-03 10:32:18 
2024-04-03 10:32:18 io.camunda.common.exception.SdkException: io.camunda.common.exception.SdkException: Response not successful: 500
2024-04-03 10:32:18     at io.camunda.common.http.DefaultHttpClient.post(DefaultHttpClient.java:146)
2024-04-03 10:32:18     at io.camunda.operate.CamundaOperateClient.searchProcessDefinitionResults(CamundaOperateClient.java:46)
2024-04-03 10:32:18     at io.camunda.connector.runtime.inbound.importer.ProcessDefinitionSearch.query(ProcessDefinitionSearch.java:64)
2024-04-03 10:32:18     at io.camunda.connector.runtime.inbound.importer.ProcessDefinitionImporter.scheduleImport(ProcessDefinitionImporter.java:52)
2024-04-03 10:32:18     at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source)
2024-04-03 10:32:18     at java.base/java.lang.reflect.Method.invoke(Unknown Source)
2024-04-03 10:32:18     at org.springframework.scheduling.support.ScheduledMethodRunnable.runInternal(ScheduledMethodRunnable.java:130)
2024-04-03 10:32:18     at org.springframework.scheduling.support.ScheduledMethodRunnable.lambda$run$2(ScheduledMethodRunnable.java:124)
2024-04-03 10:32:18     at io.micrometer.observation.Observation.observe(Observation.java:499)
2024-04-03 10:32:18     at org.springframework.scheduling.support.ScheduledMethodRunnable.run(ScheduledMethodRunnable.java:124)
2024-04-03 10:32:18     at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54)
2024-04-03 10:32:18     at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
2024-04-03 10:32:18     at java.base/java.util.concurrent.FutureTask.runAndReset(Unknown Source)
2024-04-03 10:32:18     at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
2024-04-03 10:32:18     at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
2024-04-03 10:32:18     at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
2024-04-03 10:32:18     at java.base/java.lang.Thread.run(Unknown Source)
2024-04-03 10:32:18 Caused by: io.camunda.common.exception.SdkException: Response not successful: 500
2024-04-03 10:32:18     at io.camunda.common.http.DefaultHttpClient.parseAndRetry(DefaultHttpClient.java:257)
2024-04-03 10:32:18     at io.camunda.common.http.DefaultHttpClient.post(DefaultHttpClient.java:143)
2024-04-03 10:32:18     ... 16 common frames omitted
2024-04-03 10:32:18 
```

As Operate is called here, the Operate logs should also show something:

```
2024-04-03 10:33:29 2024-04-03 08:33:29.031 ERROR 7 --- [nio-8080-exec-4] o.a.c.c.C.[.[.[.[dispatcherServlet]      : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception
2024-04-03 10:33:29 
2024-04-03 10:33:29 org.springframework.security.authentication.AuthenticationServiceException: An error occurred while attempting to decode the Jwt: Couldn't retrieve remote JWK set: org.springframework.web.client.ResourceAccessException: I/O error on GET request for "http://localhost:18080/auth/realms/camunda-platform/protocol/openid-connect/certs": Connection refused
2024-04-03 10:33:29     at org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationProvider.getJwt(JwtAuthenticationProvider.java:106) ~[spring-security-oauth2-resource-server-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationProvider.authenticate(JwtAuthenticationProvider.java:88) ~[spring-security-oauth2-resource-server-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.authentication.ProviderManager.authenticate(ProviderManager.java:182) ~[spring-security-core-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.authentication.ObservationAuthenticationManager.lambda$authenticate$1(ObservationAuthenticationManager.java:54) ~[spring-security-core-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at io.micrometer.observation.Observation.lambda$observe$4(Observation.java:544) ~[micrometer-observation-1.11.9.jar!/:1.11.9]
2024-04-03 10:33:29     at io.micrometer.observation.Observation.observeWithContext(Observation.java:603) ~[micrometer-observation-1.11.9.jar!/:1.11.9]
2024-04-03 10:33:29     at io.micrometer.observation.Observation.observe(Observation.java:544) ~[micrometer-observation-1.11.9.jar!/:1.11.9]
2024-04-03 10:33:29     at org.springframework.security.authentication.ObservationAuthenticationManager.authenticate(ObservationAuthenticationManager.java:53) ~[spring-security-core-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.oauth2.server.resource.web.authentication.BearerTokenAuthenticationFilter.doFilterInternal(BearerTokenAuthenticationFilter.java:137) ~[spring-security-oauth2-resource-server-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:107) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:93) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.web.header.HeaderWriterFilter.doHeadersAfter(HeaderWriterFilter.java:90) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:75) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.web.context.SecurityContextHolderFilter.doFilter(SecurityContextHolderFilter.java:82) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.web.context.SecurityContextHolderFilter.doFilter(SecurityContextHolderFilter.java:69) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:62) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:227) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.web.session.DisableEncodeUrlFilter.doFilterInternal(DisableEncodeUrlFilter.java:42) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.wrapFilter(ObservationFilterChainDecorator.java:240) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.web.ObservationFilterChainDecorator$AroundFilterObservation$SimpleAroundFilterObservation.lambda$wrap$0(ObservationFilterChainDecorator.java:323) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.web.ObservationFilterChainDecorator$ObservationFilter.doFilter(ObservationFilterChainDecorator.java:224) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.web.ObservationFilterChainDecorator$VirtualFilterChain.doFilter(ObservationFilterChainDecorator.java:137) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:233) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:191) ~[spring-security-web-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:113) ~[spring-web-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.springframework.web.servlet.handler.HandlerMappingIntrospector.lambda$createCacheFilter$3(HandlerMappingIntrospector.java:195) ~[spring-webmvc-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:113) ~[spring-web-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.springframework.web.filter.CompositeFilter.doFilter(CompositeFilter.java:74) ~[spring-web-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.springframework.security.config.annotation.web.configuration.WebMvcSecurityConfiguration$CompositeFilterChainProxy.doFilter(WebMvcSecurityConfiguration.java:230) ~[spring-security-config-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:352) ~[spring-web-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:268) ~[spring-web-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174) ~[tomcat-embed-core-10.1.19.jar!/:?]
2024-04-03 10:33:29     at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.19.jar!/:?]
2024-04-03 10:33:29     at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100) ~[spring-web-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174) ~[tomcat-embed-core-10.1.19.jar!/:?]
2024-04-03 10:33:29     at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.19.jar!/:?]
2024-04-03 10:33:29     at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93) ~[spring-web-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174) ~[tomcat-embed-core-10.1.19.jar!/:?]
2024-04-03 10:33:29     at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.19.jar!/:?]
2024-04-03 10:33:29     at org.springframework.web.filter.ServerHttpObservationFilter.doFilterInternal(ServerHttpObservationFilter.java:109) ~[spring-web-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174) ~[tomcat-embed-core-10.1.19.jar!/:?]
2024-04-03 10:33:29     at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.19.jar!/:?]
2024-04-03 10:33:29     at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201) ~[spring-web-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174) ~[tomcat-embed-core-10.1.19.jar!/:?]
2024-04-03 10:33:29     at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.19.jar!/:?]
2024-04-03 10:33:29     at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:167) [tomcat-embed-core-10.1.19.jar!/:?]
2024-04-03 10:33:29     at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90) [tomcat-embed-core-10.1.19.jar!/:?]
2024-04-03 10:33:29     at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:482) [tomcat-embed-core-10.1.19.jar!/:?]
2024-04-03 10:33:29     at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:115) [tomcat-embed-core-10.1.19.jar!/:?]
2024-04-03 10:33:29     at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93) [tomcat-embed-core-10.1.19.jar!/:?]
2024-04-03 10:33:29     at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74) [tomcat-embed-core-10.1.19.jar!/:?]
2024-04-03 10:33:29     at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344) [tomcat-embed-core-10.1.19.jar!/:?]
2024-04-03 10:33:29     at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:391) [tomcat-embed-core-10.1.19.jar!/:?]
2024-04-03 10:33:29     at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63) [tomcat-embed-core-10.1.19.jar!/:?]
2024-04-03 10:33:29     at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:896) [tomcat-embed-core-10.1.19.jar!/:?]
2024-04-03 10:33:29     at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1744) [tomcat-embed-core-10.1.19.jar!/:?]
2024-04-03 10:33:29     at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52) [tomcat-embed-core-10.1.19.jar!/:?]
2024-04-03 10:33:29     at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191) [tomcat-embed-core-10.1.19.jar!/:?]
2024-04-03 10:33:29     at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659) [tomcat-embed-core-10.1.19.jar!/:?]
2024-04-03 10:33:29     at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63) [tomcat-embed-core-10.1.19.jar!/:?]
2024-04-03 10:33:29     at java.lang.Thread.run(Thread.java:840) [?:?]
2024-04-03 10:33:29 Caused by: org.springframework.security.oauth2.jwt.JwtException: An error occurred while attempting to decode the Jwt: Couldn't retrieve remote JWK set: org.springframework.web.client.ResourceAccessException: I/O error on GET request for "http://localhost:18080/auth/realms/camunda-platform/protocol/openid-connect/certs": Connection refused
2024-04-03 10:33:29     at org.springframework.security.oauth2.jwt.NimbusJwtDecoder.createJwt(NimbusJwtDecoder.java:173) ~[spring-security-oauth2-jose-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.oauth2.jwt.NimbusJwtDecoder.decode(NimbusJwtDecoder.java:138) ~[spring-security-oauth2-jose-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationProvider.getJwt(JwtAuthenticationProvider.java:99) ~[spring-security-oauth2-resource-server-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     ... 82 more
2024-04-03 10:33:29 Caused by: com.nimbusds.jose.RemoteKeySourceException: Couldn't retrieve remote JWK set: org.springframework.web.client.ResourceAccessException: I/O error on GET request for "http://localhost:18080/auth/realms/camunda-platform/protocol/openid-connect/certs": Connection refused
2024-04-03 10:33:29     at com.nimbusds.jose.jwk.source.RemoteJWKSet.updateJWKSetFromURL(RemoteJWKSet.java:309) ~[nimbus-jose-jwt-9.37.2.jar!/:9.37.2]
2024-04-03 10:33:29     at com.nimbusds.jose.jwk.source.RemoteJWKSet.get(RemoteJWKSet.java:439) ~[nimbus-jose-jwt-9.37.2.jar!/:9.37.2]
2024-04-03 10:33:29     at com.nimbusds.jose.proc.JWSVerificationKeySelector.selectJWSKeys(JWSVerificationKeySelector.java:157) ~[nimbus-jose-jwt-9.37.2.jar!/:9.37.2]
2024-04-03 10:33:29     at com.nimbusds.jwt.proc.DefaultJWTProcessor.selectKeys(DefaultJWTProcessor.java:283) ~[nimbus-jose-jwt-9.37.2.jar!/:9.37.2]
2024-04-03 10:33:29     at com.nimbusds.jwt.proc.DefaultJWTProcessor.process(DefaultJWTProcessor.java:354) ~[nimbus-jose-jwt-9.37.2.jar!/:9.37.2]
2024-04-03 10:33:29     at com.nimbusds.jwt.proc.DefaultJWTProcessor.process(DefaultJWTProcessor.java:303) ~[nimbus-jose-jwt-9.37.2.jar!/:9.37.2]
2024-04-03 10:33:29     at org.springframework.security.oauth2.jwt.NimbusJwtDecoder.createJwt(NimbusJwtDecoder.java:158) ~[spring-security-oauth2-jose-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.oauth2.jwt.NimbusJwtDecoder.decode(NimbusJwtDecoder.java:138) ~[spring-security-oauth2-jose-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationProvider.getJwt(JwtAuthenticationProvider.java:99) ~[spring-security-oauth2-resource-server-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     ... 82 more
2024-04-03 10:33:29 Caused by: java.io.IOException: org.springframework.web.client.ResourceAccessException: I/O error on GET request for "http://localhost:18080/auth/realms/camunda-platform/protocol/openid-connect/certs": Connection refused
2024-04-03 10:33:29     at org.springframework.security.oauth2.jwt.NimbusJwtDecoder$JwkSetUriJwtDecoderBuilder$RestOperationsResourceRetriever.getResponse(NimbusJwtDecoder.java:491) ~[spring-security-oauth2-jose-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.oauth2.jwt.NimbusJwtDecoder$JwkSetUriJwtDecoderBuilder$RestOperationsResourceRetriever.retrieveResource(NimbusJwtDecoder.java:478) ~[spring-security-oauth2-jose-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at com.nimbusds.jose.jwk.source.RemoteJWKSet.updateJWKSetFromURL(RemoteJWKSet.java:307) ~[nimbus-jose-jwt-9.37.2.jar!/:9.37.2]
2024-04-03 10:33:29     at com.nimbusds.jose.jwk.source.RemoteJWKSet.get(RemoteJWKSet.java:439) ~[nimbus-jose-jwt-9.37.2.jar!/:9.37.2]
2024-04-03 10:33:29     at com.nimbusds.jose.proc.JWSVerificationKeySelector.selectJWSKeys(JWSVerificationKeySelector.java:157) ~[nimbus-jose-jwt-9.37.2.jar!/:9.37.2]
2024-04-03 10:33:29     at com.nimbusds.jwt.proc.DefaultJWTProcessor.selectKeys(DefaultJWTProcessor.java:283) ~[nimbus-jose-jwt-9.37.2.jar!/:9.37.2]
2024-04-03 10:33:29     at com.nimbusds.jwt.proc.DefaultJWTProcessor.process(DefaultJWTProcessor.java:354) ~[nimbus-jose-jwt-9.37.2.jar!/:9.37.2]
2024-04-03 10:33:29     at com.nimbusds.jwt.proc.DefaultJWTProcessor.process(DefaultJWTProcessor.java:303) ~[nimbus-jose-jwt-9.37.2.jar!/:9.37.2]
2024-04-03 10:33:29     at org.springframework.security.oauth2.jwt.NimbusJwtDecoder.createJwt(NimbusJwtDecoder.java:158) ~[spring-security-oauth2-jose-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.oauth2.jwt.NimbusJwtDecoder.decode(NimbusJwtDecoder.java:138) ~[spring-security-oauth2-jose-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationProvider.getJwt(JwtAuthenticationProvider.java:99) ~[spring-security-oauth2-resource-server-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     ... 82 more
2024-04-03 10:33:29 Caused by: org.springframework.web.client.ResourceAccessException: I/O error on GET request for "http://localhost:18080/auth/realms/camunda-platform/protocol/openid-connect/certs": Connection refused
2024-04-03 10:33:29     at org.springframework.web.client.RestTemplate.createResourceAccessException(RestTemplate.java:890) ~[spring-web-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:870) ~[spring-web-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.springframework.web.client.RestTemplate.exchange(RestTemplate.java:705) ~[spring-web-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.springframework.security.oauth2.jwt.NimbusJwtDecoder$JwkSetUriJwtDecoderBuilder$RestOperationsResourceRetriever.getResponse(NimbusJwtDecoder.java:488) ~[spring-security-oauth2-jose-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.oauth2.jwt.NimbusJwtDecoder$JwkSetUriJwtDecoderBuilder$RestOperationsResourceRetriever.retrieveResource(NimbusJwtDecoder.java:478) ~[spring-security-oauth2-jose-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at com.nimbusds.jose.jwk.source.RemoteJWKSet.updateJWKSetFromURL(RemoteJWKSet.java:307) ~[nimbus-jose-jwt-9.37.2.jar!/:9.37.2]
2024-04-03 10:33:29     at com.nimbusds.jose.jwk.source.RemoteJWKSet.get(RemoteJWKSet.java:439) ~[nimbus-jose-jwt-9.37.2.jar!/:9.37.2]
2024-04-03 10:33:29     at com.nimbusds.jose.proc.JWSVerificationKeySelector.selectJWSKeys(JWSVerificationKeySelector.java:157) ~[nimbus-jose-jwt-9.37.2.jar!/:9.37.2]
2024-04-03 10:33:29     at com.nimbusds.jwt.proc.DefaultJWTProcessor.selectKeys(DefaultJWTProcessor.java:283) ~[nimbus-jose-jwt-9.37.2.jar!/:9.37.2]
2024-04-03 10:33:29     at com.nimbusds.jwt.proc.DefaultJWTProcessor.process(DefaultJWTProcessor.java:354) ~[nimbus-jose-jwt-9.37.2.jar!/:9.37.2]
2024-04-03 10:33:29     at com.nimbusds.jwt.proc.DefaultJWTProcessor.process(DefaultJWTProcessor.java:303) ~[nimbus-jose-jwt-9.37.2.jar!/:9.37.2]
2024-04-03 10:33:29     at org.springframework.security.oauth2.jwt.NimbusJwtDecoder.createJwt(NimbusJwtDecoder.java:158) ~[spring-security-oauth2-jose-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.oauth2.jwt.NimbusJwtDecoder.decode(NimbusJwtDecoder.java:138) ~[spring-security-oauth2-jose-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationProvider.getJwt(JwtAuthenticationProvider.java:99) ~[spring-security-oauth2-resource-server-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     ... 82 more
2024-04-03 10:33:29 Caused by: java.net.ConnectException: Connection refused
2024-04-03 10:33:29     at sun.nio.ch.Net.connect0(Native Method) ~[?:?]
2024-04-03 10:33:29     at sun.nio.ch.Net.connect(Net.java:579) ~[?:?]
2024-04-03 10:33:29     at sun.nio.ch.Net.connect(Net.java:568) ~[?:?]
2024-04-03 10:33:29     at sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:593) ~[?:?]
2024-04-03 10:33:29     at java.net.Socket.connect(Socket.java:633) ~[?:?]
2024-04-03 10:33:29     at java.net.Socket.connect(Socket.java:583) ~[?:?]
2024-04-03 10:33:29     at sun.net.NetworkClient.doConnect(NetworkClient.java:183) ~[?:?]
2024-04-03 10:33:29     at sun.net.www.http.HttpClient.openServer(HttpClient.java:533) ~[?:?]
2024-04-03 10:33:29     at sun.net.www.http.HttpClient.openServer(HttpClient.java:638) ~[?:?]
2024-04-03 10:33:29     at sun.net.www.http.HttpClient.<init>(HttpClient.java:281) ~[?:?]
2024-04-03 10:33:29     at sun.net.www.http.HttpClient.New(HttpClient.java:386) ~[?:?]
2024-04-03 10:33:29     at sun.net.www.http.HttpClient.New(HttpClient.java:408) ~[?:?]
2024-04-03 10:33:29     at sun.net.www.protocol.http.HttpURLConnection.getNewHttpClient(HttpURLConnection.java:1309) ~[?:?]
2024-04-03 10:33:29     at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1242) ~[?:?]
2024-04-03 10:33:29     at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1128) ~[?:?]
2024-04-03 10:33:29     at sun.net.www.protocol.http.HttpURLConnection.connect(HttpURLConnection.java:1057) ~[?:?]
2024-04-03 10:33:29     at org.springframework.http.client.SimpleBufferingClientHttpRequest.executeInternal(SimpleBufferingClientHttpRequest.java:75) ~[spring-web-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.springframework.http.client.AbstractBufferingClientHttpRequest.executeInternal(AbstractBufferingClientHttpRequest.java:48) ~[spring-web-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.springframework.http.client.AbstractClientHttpRequest.execute(AbstractClientHttpRequest.java:66) ~[spring-web-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:864) ~[spring-web-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.springframework.web.client.RestTemplate.exchange(RestTemplate.java:705) ~[spring-web-6.0.17.jar!/:6.0.17]
2024-04-03 10:33:29     at org.springframework.security.oauth2.jwt.NimbusJwtDecoder$JwkSetUriJwtDecoderBuilder$RestOperationsResourceRetriever.getResponse(NimbusJwtDecoder.java:488) ~[spring-security-oauth2-jose-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.oauth2.jwt.NimbusJwtDecoder$JwkSetUriJwtDecoderBuilder$RestOperationsResourceRetriever.retrieveResource(NimbusJwtDecoder.java:478) ~[spring-security-oauth2-jose-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at com.nimbusds.jose.jwk.source.RemoteJWKSet.updateJWKSetFromURL(RemoteJWKSet.java:307) ~[nimbus-jose-jwt-9.37.2.jar!/:9.37.2]
2024-04-03 10:33:29     at com.nimbusds.jose.jwk.source.RemoteJWKSet.get(RemoteJWKSet.java:439) ~[nimbus-jose-jwt-9.37.2.jar!/:9.37.2]
2024-04-03 10:33:29     at com.nimbusds.jose.proc.JWSVerificationKeySelector.selectJWSKeys(JWSVerificationKeySelector.java:157) ~[nimbus-jose-jwt-9.37.2.jar!/:9.37.2]
2024-04-03 10:33:29     at com.nimbusds.jwt.proc.DefaultJWTProcessor.selectKeys(DefaultJWTProcessor.java:283) ~[nimbus-jose-jwt-9.37.2.jar!/:9.37.2]
2024-04-03 10:33:29     at com.nimbusds.jwt.proc.DefaultJWTProcessor.process(DefaultJWTProcessor.java:354) ~[nimbus-jose-jwt-9.37.2.jar!/:9.37.2]
2024-04-03 10:33:29     at com.nimbusds.jwt.proc.DefaultJWTProcessor.process(DefaultJWTProcessor.java:303) ~[nimbus-jose-jwt-9.37.2.jar!/:9.37.2]
2024-04-03 10:33:29     at org.springframework.security.oauth2.jwt.NimbusJwtDecoder.createJwt(NimbusJwtDecoder.java:158) ~[spring-security-oauth2-jose-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.oauth2.jwt.NimbusJwtDecoder.decode(NimbusJwtDecoder.java:138) ~[spring-security-oauth2-jose-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     at org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationProvider.getJwt(JwtAuthenticationProvider.java:99) ~[spring-security-oauth2-resource-server-6.1.7.jar!/:6.1.7]
2024-04-03 10:33:29     ... 82 more
2024-04-03 10:33:29 
```

This stacktrace indicates that spring security is misconfigured for Operate.

On inspecting the container, the following env variables show up:

```
			"SPRING_SECURITY_OAUTH2_RESOURCESERVER_JWT_ISSUER_URI=http://localhost:18080/auth/realms/camunda-platform",
			"SPRING_SECURITY_OAUTH2_RESOURCESERVER_JWT_JWK_SET_URI=http://localhost:18080/auth/realms/camunda-platform/protocol/openid-connect/certs",
```

Both point to localhost, which is not correct (obviously, as the docker container is running inside a dedicated network)

--- a/docker-compose.yaml
+++ b/docker-compose.yaml
@@ -69,8 +69,8 @@ services:
       - CAMUNDA_OPERATE_IDENTITY_CLIENTSECRET=XALaRPl5qwTEItdwCMiPS62nVpKs7dL7
       - CAMUNDA_OPERATE_IDENTITY_AUDIENCE=operate-api
       - CAMUNDA_OPERATE_MULTITENANCY_ENABLED=${MULTI_TENANCY_ENABLED}
-      - SPRING_SECURITY_OAUTH2_RESOURCESERVER_JWT_ISSUER_URI=http://${HOST}:18080/auth/realms/camunda-platform
-      - SPRING_SECURITY_OAUTH2_RESOURCESERVER_JWT_JWK_SET_URI=http://${HOST}:18080/auth/realms/camunda-platform/protocol/openid-connect/certs
+      - SPRING_SECURITY_OAUTH2_RESOURCESERVER_JWT_ISSUER_URI=http://keycloak:8080/auth/realms/camunda-platform
+      - SPRING_SECURITY_OAUTH2_RESOURCESERVER_JWT_JWK_SET_URI=http://keycloak:8080/auth/realms/camunda-platform/protocol/openid-connect/certs
       - CAMUNDA_OPERATE_IDENTITY_RESOURCEPERMISSIONSENABLED=${RESOURCE_AUTHORIZATIONS_ENABLED}
       - management.endpoints.web.exposure.include=health
       - management.endpoint.health.probes.enabled=true
@@ -114,8 +114,8 @@ services:
       - CAMUNDA_TASKLIST_IDENTITY_CLIENTSECRET=XALaRPl5qwTEItdwCMiPS62nVpKs7dL7
       - CAMUNDA_TASKLIST_IDENTITY_AUDIENCE=tasklist-api
       - CAMUNDA_TASKLIST_MULTITENANCY_ENABLED=${MULTI_TENANCY_ENABLED}
-      - SPRING_SECURITY_OAUTH2_RESOURCESERVER_JWT_ISSUER_URI=http://${HOST}:18080/auth/realms/camunda-platform
-      - SPRING_SECURITY_OAUTH2_RESOURCESERVER_JWT_JWK_SET_URI=http://${HOST}:18080/auth/realms/camunda-platform/protocol/openid-connect/certs
+      - SPRING_SECURITY_OAUTH2_RESOURCESERVER_JWT_ISSUER_URI=http://keycloak:8080/auth/realms/camunda-platform
+      - SPRING_SECURITY_OAUTH2_RESOURCESERVER_JWT_JWK_SET_URI=http://keycloak:8080/auth/realms/camunda-platform/protocol/openid-connect/certs
       - CAMUNDA_TASKLIST_IDENTITY_RESOURCE_PERMISSIONS_ENABLED=${RESOURCE_AUTHORIZATIONS_ENABLED}
       - management.endpoints.web.exposure.include=health
       - management.endpoint.health.probes.enabled=true

